test/gc_test.go

```diff
@@ -0,0 +1,210 @@
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gadget-inc/dateilager/internal/auth"
+	"github.com/gadget-inc/dateilager/internal/pb"
+	util "github.com/gadget-inc/dateilager/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGcProjectRemovesObjectsAndContent(t *testing.T) {
+	tc := util.NewTestCtx(t, auth.Admin)
+	defer tc.Close()
+
+	writeProject(tc, 1, 4)
+	writeObject(tc, 1, 1, i(2), "/a", "a v1")
+	writeObject(tc, 1, 2, i(3), "/b", "b v2")
+	writeObject(tc, 1, 3, i(4), "/b", "b v3")
+	writeObject(tc, 1, 3, nil, "/c", "c v3")
+	writeObject(tc, 1, 4, nil, "/d", "d v4")
+
+	fs := tc.FsApi()
+
+	response, err := fs.GcProject(tc.Context(), &pb.GcProjectRequest{
+		Project:      1,
+		KeepVersions: 1,
+	})
+	require.NoError(t, err, "fs.GcProject")
+
+	assert.Equal(t, int64(1), response.Project, "Gc result project")
+	assert.Equal(t, int64(2), response.Count, "Gc result count")
```

Comment:
Just to make sure I'm understanding the tests correctly: the two objects being GC-ed here are

```
riteObject(tc, 1, 1, i(2), "/a", "a v1")
riteObject(tc, 1, 2, i(3), "/b", "b v2")
```

But because we keep a single version, versions 3 and 4 are kept, so we keep this one:

```
writeObject(tc, 1, 3, i(4), "/b", "b v3")
```

Reply:
Yeah that's correct.

Scott had the same question though, we technically keep 2 versions here because we keep things that ended in v4.

I don't know that it's strictly necessary to do so, but keeping an extra version also isn't super problematic. Let me see if I can change that.